aci/resource_aci_bgppeerp.go

```diff
@@ -633,28 +637,35 @@ func resourceAciBgpPeerConnectivityProfileUpdate(ctx context.Context, d *schema.
 			bgpLocalAsnPAttr.LocalAsn = LocalAsn.(string)
 		}
 
-		bgpLocalAsnP := models.NewLocalAutonomousSystemProfile(fmt.Sprintf("localasn"), PeerConnectivityProfileDn, desc, bgpLocalAsnPAttr)
-		bgpLocalAsnP.Status = "modified"
+		if _, ok := d.GetOk("local_asn"); ok {
+			bgpLocalAsnP := models.NewLocalAutonomousSystemProfile(fmt.Sprintf("localasn"), PeerConnectivityProfileDn, desc, bgpLocalAsnPAttr)
 
-		err = aciClient.Save(bgpLocalAsnP)
+			err = aciClient.Save(bgpLocalAsnP)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	if d.HasChange("as_number") {
+
+		err := aciClient.DeleteByDn(dn+"/as", "bgpAsP")
```

Comment:
should we construct/format the string consistently in our code with fmt.Sprintf("%s/as", dn)?

Reply:
should we point to models.BgpaspClassName in bgp_as_p.go of aci client?